File: parser/src/main/java/com/github/nsc/de/shake/parser/node/factor/CharacterNode.java
package com.github.nsc.de.shake.parser.node.factor;

import com.github.nsc.de.shake.parser.node.ValuedNode;
import com.github.nsc.de.shake.util.characterinput.position.PositionMap;

public class CharacterNode extends ValuedNode {

    private final char value;

    public CharacterNode(PositionMap map, char value) {
        super(map);
        this.value = value;
    }

    public char getValue() {
        return value;
    }
}

File: parser/src/main/kotlin/com/github/nsc/de/shake/parser/node/factor/CharacterNode.kt
package com.github.nsc.de.shake.parser.node.factor

import com.github.nsc.de.shake.parser.node.ValuedNode
import com.github.nsc.de.shake.util.characterinput.position.PositionMap

class CharacterNode(map: PositionMap, val value: Char) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "CharacterNode", "value" to "$value")
}